lib/Factory/PostFactory.php

```diff
@@ -71,31 +73,48 @@ protected function from_wp_query(WP_Query $query) : Iterable {
 	}
 
 	protected function get_post_class(WP_Post $post) : string {
-    // Get the user-configured Class Map
-    $map = apply_filters( 'timber/post/classmap', [
-      'post'       => Post::class,
-      'page'       => Post::class,
-      // @todo special logic for attachments?
-      'attachment' => Attachment::class,
-    ] );
+		// Get the user-configured Class Map
+		$map = apply_filters( 'timber/post/classmap', [
+			'post'       => Post::class,
+			'page'       => Post::class,
+			// Apply special logic for attachments.
+			'attachment' => function(WP_Post $attachment) {
+				return $this->is_image($attachment) ? Image::class : Attachment::class;
+			},
+		] );
 
 		$class = $map[$post->post_type] ?? null;
 
-    // If class is a callable, call it to get the actual class name
-    if (is_callable($class)) {
-      $class = $class($post);
-    }
+		// If class is a callable, call it to get the actual class name
+		if (is_callable($class)) {
+			$class = $class($post);
+		}
 
-    // If we don't have a post class by now, fallback on the default class
-    return $class ?? Post::class;
+		// If we don't have a post class by now, fallback on the default class
+		return $class ?? Post::class;
 	}
 
-  protected function build(WP_Post $post) : CoreInterface {
+	protected function is_image(WP_Post $post) {
+		$src   = wp_get_attachment_url( $post->ID );
+		$check = wp_check_filetype( PathHelper::basename( $src ) );
```

Comment:
If we do that I think we'd also need to change the `timber/post/image_extensions` filter to not only care about the extensions but also about the mime type, so users retain control over which attachments are considered images or not. So rather than a flat array of extensions, we'd run the filter earlier and usage would look like:

```php
add_filter('timber/post/image_filetypes', function(array $ext_regex_to_mime) {
  return array_merge($ext_regex_to_mime, [
    '/ext/' => 'image/something',
  ]);
});
```

This is more mental load on the user so I'm curious what you're concerned about here. :) Security? Performance?

Reply:
For context, I was going off of what @jarednova said in an earlier comment:

> there's an Image::is_image() function we could use to share a common definition (and be extensible/filterable)

Hence the need for an extra filter. That said, WP hard-codes this list in [`wp_attachment_is()`](https://developer.wordpress.org/reference/functions/wp_attachment_is/). So we could side-step this whole issue and just use that. It's not my favorite option because it doesn't support WEBP but just throwing that out there.

Reply:
@acobster I was concerned about performance. On a site I recently worked on, [Blackfire](https://blackfire.io) showed that `wp_check_filetype()` was eating performance when I tried to load a lot of images on one page. I figured that when I pass the mime types I want to check for, then the `get_allowed_mime_types()` function inside `wp_check_filetype()` function as well as the `upload_mimes` filter are skipped. There was probably something else hooking into `upload_mimes` that used a lot of performance. But I thought since I’m only checking for images, I could do it like this:

```php
$mime_types = [
    'jpg|jpeg|jpe' => 'image/jpeg',
    'gif'          => 'image/gif',
    'png'          => 'image/png',
];

$check = wp_check_filetype( $filename, $mime_types );
```

But you are right, it gets trickier if we want to support WebP, because the WebP mime type needs to be added through the `upload_mimes` filter. If there should ever be a performance problem in the future, we can think about a solution then.

> For context, I was going off of what @jarednova said in an earlier comment:
> 
> > there's an Image::is_image() function we could use to share a common definition (and be extensible/filterable)

I’m all for that!

> That said, WP hard-codes this list in [`wp_attachment_is()`](https://developer.wordpress.org/reference/functions/wp_attachment_is/). So we could side-step this whole issue and just use that. It's not my favorite option because it doesn't support WEBP but just throwing that out there.

Yeah, not my favorite option either if it doesn’t support WebP.